### PR TITLE
Ic pagination cypress

### PR DESCRIPTION
**Pull request by mmb675**
Complete Cypress Pagination Ticket: End-to-End, Regression, and Accessibility Tests.

## Summary of the changes
Tell us, in a few words, what the changes are.

## Related issue
#1546

### CODE REVIEW

**Review by gd2910 — CHANGES_REQUESTED** [review]

**Review by jd239 — DISMISSED** [review]
A few comments

File: packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+describe("IcPagination e2e, A11y and visual regression tests", () => {
+  beforeEach(() => {
+    cy.injectAxe();
+  });
+
+  afterEach(() => {
+    cy.task("generateReport");
+  });
+
+  it("reanders simple type pagination", () => {
```
Comment (gd2910) [inline packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx:18]:
```suggestion
  it("renders simple type pagination", () => {
```

File: packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+    cy.findShadowEl("ic-pagination", "ic-button").each(($button) => {
+      $button.focus();
+    });
+    cy.compareSnapshot({
+      name: "focus_simple_pagination",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("renader complex type pagination", () => {
```
Comment (gd2910) [inline packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx:35]:
```suggestion
  it("renders complex type pagination", () => {
```

File: packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,468 @@
+/// <reference types="Cypress" />
+
+import React from "react";
+import { IcPagination } from "../../components";
+import { mount } from "cypress/react";
+import { NOT_EXIST, NOT_HAVE_CLASS } from "../utils/constants";
+import { setThresholdBasedOnEnv } from "../../../cypress/utils/helpers";
+
+describe("IcPagination e2e, A11y and visual regression tests", () => {
```
Comment (gd2910) [inline packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx:9]:
```suggestion
describe("IcPagination", () => {
```

File: packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx
Comment (gd2910) [inline packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx:1]: Most snapshots need a top margin to show the full focused element.

Comment (gd2910) [inline packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx:1]: When running locally I'm getting a lot of visual regression failures. On checking the 'Image Diff report' there are a lot of variances in passing and failing tests. In addition to above comments from @jd239, all of the threshold numbers are now +0.02?

Comment (mmb675) [reply packages/react/src/component-tests/IcPagination/IcPagination.cy.tsx:1]: it is now 0.02  change to new screenshot test.

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by gd2910 — APPROVED** [review]

**Pull request merged**